File: swift-algorithm/swift-algorithm/LeetCode/LeetCode/125_valid_palindrome.swift
//
//  main.swift
//  swift-algorithm
//
//  Created by 전판근 on 2020/12/25.
//

import Foundation

class Solution {
    func isPalindrome(_ s: String) -> Bool {
        
        let lowerString = s.lowercased()
        let alnumString = lowerString.replacingOccurrences(of: "[^a-z0-9]", with: "", options: .regularExpression)
        let reversedString = String(alnumString.reversed())
        
        return alnumString == reversedString
        
    }
}

//Solution.isPalindrome("A man, a plan, a canal: Panama")

var solution = Solution()
let sol = solution.isPalindrome("A man, a plan, a canal: Panama")

print(sol)

File: python/LeetCode/125_Valid_Palindrome.py
import re


class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()

        s = re.sub('[^a-z0-9]', '', s)

        return s == s[:: -1]


# Input
S = Solution()

input = input()
S.isPalindrome(input)
